reflect.go

```diff
@@ -951,11 +951,9 @@ func (r *Reflector) reflectFieldName(f reflect.StructField) (string, bool, bool,
 
 	// field anonymous but without json tag should be inherited by current type
 	if f.Anonymous && !exist {
-		if !r.YAMLEmbeddedStructs {
+		if !r.YAMLEmbeddedStructs && f.Type.Kind() == reflect.Struct {
```

Comment:
I'm not sure what to think about this.

The `YAMLEmbeddedStructs` is something I haven't managed to understand. Even reading the original PR I still don't get it: https://github.com/alecthomas/jsonschema/pull/74

I do think however the struct check should be on it's own line, even if that results in duplicated code for the sake of clarity.

What do you think?

Reply:
It seems to me that the `YAMLEmbeddedStructs` feature is mostly about disabling embedding, and the association with YAML is probably unnecessary.

Embedding semantics _should_ be customizable through options to accommodate everyone's needs. In my opinion some more work is needed to improve the configuration API in this area.

I would be happy to make additional changes along those lines.

Reply:
Okay, I get your point now and agree. The `YAML` part and naming in general is very confusing, in effect, it does the reverse of what it suggests.

I'd suggest an alternative approach:
 * Remove the `YAMLEmbeddedStructs` option.
 * Embed Anonymous structs by default, i.e. the same as what the current `YAMLEmbeddedStructs = false` does.
 * Continue to ignore any other anonymous type, like string, map, etc.
 * If the Anonymous type (string, map, struct, etc.) has a `json:"name"` tag, then we treat it as a property.

Would that work for your use-case?

Reply:
Yes, that would work for me, but it doesn't seem like the best solution. I would expect this tool to follow Go's conventions for serialization by default, and offer configuration options to customize the behavior as needed. I think that would require a breaking change because the current inlining behavior is at odds with Go's normal embedding behavior. If you're open to releasing a new major version, then please consider the following change list:

- Remove the `YAMLEmbeddedStructs` option.
- Do not inline anonymous structs by default, i.e. the same as what the current `YAMLEmbeddedStructs = true` does.
- Embed all anonymous types, for consistency with Go and with the struct embedding behavior.
- If any anonymous type has a `json:"name"` tag, then we change the name of the property in the schema.
- If an anonymous struct field has a `jsonschema:"inline"` tag, then inline the fields of the struct, i.e. the same as what the current `YAMLEmbeddedStructs = false` / default behavior does.

By releasing a new module version, users would continue to use the current behavior until they `go get` to the next major version. To do this we only need to update the module declaration in `go.mod`:

```
module github.com/invopop/jsonschema/v2 v2.0.0
```

Reply:
Sorry, I just left a comment that was wrong so I deleted it. Here is my updated reply:

Yes, that would work for me, but it doesn't seem like the best solution because it doesn't follow Go's conventions for serialization. I would tweak your suggestions such that all anonymous types are embedded by default, and, since this is a breaking change, we would release a new major version (by changing the module declaration to `module github.com/invopop/jsonschema/v2 v2.0.0` which will require developers to `go get` the package and opt-in to the changes). This will not disrupt anyone's projects and it gives us the ideal outcome :)

Reply:
@sbward I need to fix the handling of root non-struct objects, so managed to make a proposal for this. Please see #23 and let me know if you think that would help for your use-case. Cheers.